Superk SN alert schema

hmenjo opened the pull request:
<!-- IMPORTANT!!! Aside from typographical corrections and minor changes, please consider creating an Issue before making a Pull Request. -->

# Description

Schemas for SuperK supernova notice. Two schemas are prepared for real and test notices. We plan to generate test  notice periodically like one per month because the real supernova event is super rare. Actually the format  of test notice are same as for real.

jracusin commented on gcn/notices/superk/sn_alert.real.example.json:
```diff
@@ -0,0 +1,21 @@
+{
+  "$schema": "https://gcn.nasa.gov/schema/main/gcn/notices/superk/sn_alert.real.schema.json",
+  "mission": "SK",
```
Please spell out Super-Kamiokande

hmenjo replied:
I set the mission name as "superk". Can I use the different one with it for this mission field ? 
If OK, I prefer to fill the official name "Super-Kamiokande" (but too long) to this field.

jracusin replied:
We would also prefer the full official name of the mission here.  The topic name can be shorter "superk", but the mission can be the full name.

jracusin commented on gcn/notices/superk/sn_alert.test.schema.json:
```diff
@@ -0,0 +1,33 @@
+  "$id": "https://gcn.nasa.gov/schema/main/gcn/notices/superk/sn_alert.test.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "type": "object",
+  "title": "Test of Super-Kamiokande Supernova Alert",
```
If the test and real schema are identical, they can use the same schema, but have different topics and examples.

hmenjo replied:
Thank you for your suggestion. I will change to load only one schema file for both real and test.

Vidushi-GitHub commented on gcn/notices/superk/sn_alert.test.example.json:
```diff
@@ -0,0 +1,21 @@
+  "alert_type": "initial",
+  "trigger_time": "2023-08-01T00:00:01.000000Z",
+  "processed_sample": "full",
+  "processed_pipeline": "snwatch",
```
Please use only `pipeline` word instead, if possible use enum list.

hmenjo replied:
OK, I will change the name to "pipeline" and the type to enum.

Vidushi-GitHub reviewed (commented):
IceCube and KM3Net came up with generic neutrino schema.
Please check it out, in case you wanna use it: https://github.com/nasa-gcn/gcn-schema/blob/main/gcn/notices/neutrino/Alert.schema.json

hmenjo commented:
Thank you again for your suggestions. 
I have pushed my modifications following your suggestions. 
I did not include the generic neutrino schema because our notice is for lower energy neutrinos and does not fit the schema for high energy neutrino alerts.

Vidushi-GitHub commented on gcn/notices/superk/sn_alert.real.example.json:
```diff
@@ -0,0 +1,21 @@
+{
+  "$schema": "https://gcn.nasa.gov/schema/main/gcn/notices/superk/sn_alert.schema.json",
+  "mission": "Super-Kamiokande",
+  "id": ["sk230801"],
```
Please use single string, instead of array, event core schema has been updated for it.

hmenjo replied:
OK, I have changed the type of "id" to single string.

Vidushi-GitHub commented on gcn/notices/superk/sn_alert.schema.json:
```diff
@@ -0,0 +1,33 @@
+    },
+    "processed_sample": {
+      "enum": ["partial", "full"],
+      "description": "The sample size used for this analysis process. In case of nearby supernova, because of huge statistics the initial alert will be published based on the analysis result of partial sample."
```
```suggestion
      "description": "The sample size used for this analysis process. In case of a nearby supernova, because of huge statistics the initial alert will be published based on the analysis result of partial sample."
```

hmenjo replied:
Thank you for your careful check. I have added "a".

Vidushi-GitHub reviewed (approved):
After latest feedback, schema looks fine to me.

Vidushi-GitHub commented:
Hi @hmenjo, could you please open one more PR (or, I can open for you) for updating the mission page (https://gcn.nasa.gov/missions/sksn) for new GCN JSON Notices?

hmenjo commented on gcn/notices/superk/sn_alert.schema.json:
```diff
@@ -0,0 +1,33 @@
+    },
+    "pipeline": {
+      "enum": ["snwatch", "wit"],
+      "description": "Pipleline used for this analysis process."
```
Thanks. I corrected it.

hmenjo commented:
I have pushed my modifications following your suggestions. 
Then I am going to update the document for this superk notice.  Just confirmation, the file which I should modify is /app/routes/missions.sksn/route.mdx on the gcn web page repository, right ?

Vidushi-GitHub commented:
> I have pushed my modifications following your suggestions. Then I am going to update the document for this superk notice. Just confirmation, the file which I should modify is /app/routes/missions.sksn/route.mdx on the gcn web page repository, right ?

Thanks, looks good.
Right, mission page is /app/routes/missions.sksn/route.mdx.

jracusin commented:
How are you indicating a real versus a test event?  The examples look identical.

You can either have them as separate topics:
gcn.notices.superk.sn_alert.test
gcn.notices.superk.sn_alert.real

Alternatively, you could Indicate by some parameter in the notice.  We'd recommend separate topics.

gpronost commented:
Wouldn't test events be send on the test network test.gcn.nasa.gov?
Or should we have test events on the main network too? (in this case, I agree separate topics will be better)

hmenjo commented:
> How are you indicating a real versus a test event? The examples look identical.
> 
> You can either have them as separate topics: gcn.notices.superk.sn_alert.test gcn.notices.superk.sn_alert.real
> 
> Alternatively, you could Indicate by some parameter in the notice. We'd recommend separate topics.

We plan to use the separate topics for test and real as you recommended (Actually It was suggested to me in the past discussion.).  Actually the examples are same in this moment. Do you think that it is better to have a description to identify the test and real in the examples ?

jracusin commented:
> > How are you indicating a real versus a test event? The examples look identical.
> > You can either have them as separate topics: gcn.notices.superk.sn_alert.test gcn.notices.superk.sn_alert.real
> > Alternatively, you could Indicate by some parameter in the notice. We'd recommend separate topics.
> 
> We plan to use the separate topics for test and real as you recommended (Actually It was suggested to me in the past discussion.). Actually the examples are same in this moment. Do you think that it is better to have a description to identify the test and real in the examples ?

Having test and real events as separate topics clearly differentiates them, though it would be useful for the consumers to have some indication in the message.  We suggest you use `"alert_tense": "test"` from https://github.com/nasa-gcn/gcn-schema/blob/main/gcn/notices/core/Alert.schema.json.

jracusin commented:
> Wouldn't test events be send on the test network test.gcn.nasa.gov? Or should we have test events on the main network too? (in this case, I agree separate topics will be better)

The test kafka broker is more for testing out streams before they go live.  Even in GCN Classic, SuperK sends out a regular test message type, given that the real events are exceedingly rare.  With them being separate Kafka topics, consumers can choose whether or not to receive them.

hmenjo commented:
> Having test and real events as separate topics clearly differentiates them, though it would be useful for the consumers to have some indication in the message. We suggest you use `"alert_tense": "test"` from https://github.com/nasa-gcn/gcn-schema/blob/main/gcn/notices/core/Alert.schema.json.

Thank you very much for the suggestion. I have changed "alert_tense" to "test" in example for test. 

The preparation for web documentation is on-going. Hopefully I will open the pull request for it soon.

jracusin commented:
Suggested additions:

messenger (core.reporter): "neutrino"

record_number (core.reporter)- if you issue updates to the same event, this allows you to number the subsequent notices in case of multiple updates starting with initial as 1.

hmenjo commented on gcn/notices/superk/sn_alert.schema.json:
```diff
@@ -0,0 +1,33 @@
+    },
+    "n_ibd_events": {
+      "type": "number",
+      "description": "The number of tagged inverse beta decay events in the processed sample (refer 'n_events')"
```
Done

hmenjo commented:
Thank you very much again for your checks and suggestions. 

> messenger (core.reporter): "neutrino"
> record_number (core.reporter)- if you issue updates to the same event, this allows you to number the subsequent notices in case of multiple updates starting with initial as 1.

I have added them.

In addition, I changed the valuable name "rate_duration" to "detection_interval" as suggested in the pull request for webpage description.

hmenjo commented:
I have updated a few things;
- added new valuable "trigger_number", which is an event ID used in GCN Classic.
- changed the example of event ID.

We will keep our notice on GCN Classic even after starting this new notice for follow-up observers who already implemented our notice with GCN Classic. So we thought this "trigger_number" might be useful for them to identify the same event.

Vidushi-GitHub commented on gcn/notices/superk/sn_alert.schema.json:
```diff
@@ -0,0 +1,42 @@
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "type": "object",
+  "title": "Super-Kamiokande Supernova Alert",
+  "description": "Supernova alert from Super Kamiokande",
```
```suggestion
  "description": "Supernova alert from Super-Kamiokande",
```
